packages/base-map/src/index.js

```diff
@@ -0,0 +1,271 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { LayersControl, Map, Popup, TileLayer } from "react-leaflet";
+import { latlngType } from "@opentripplanner/core-utils/lib/types";
+import L from "leaflet";
+
+// eslint-disable-next-line func-names
+L.Evented.addInitHook(function() {
+  if (this) this.singleClickTimeout = null;
+  this.on("click", this.scheduleSingleClick, this);
+  this.on("dblclick dragstart zoomstart", this.cancelSingleClick, this);
+});
+
+L.Evented.include({
+  cancelSingleClick() {
+    // This timeout is key to workaround an issue where double-click events
+    // are fired in this order on some touch browsers: ['click', 'dblclick', 'click']
+    // instead of ['click', 'click', 'dblclick']
+    setTimeout(this.clearSingleClickTimeout.bind(this), 0);
+  },
+
+  scheduleSingleClick(e) {
+    this.clearSingleClickTimeout();
+
+    this.singleClickTimeout = setTimeout(
+      this.fireSingleClick.bind(this, e),
+      this.options.singleClickTimeout || 500
+    );
+  },
+
+  fireSingleClick(e) {
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.originalEvent._stopped) {
+      this.fire("singleclick", L.Util.extend(e, { type: "singleclick" }));
+    }
+  },
+
+  clearSingleClickTimeout() {
+    if (this.singleClickTimeout !== null) {
+      clearTimeout(this.singleClickTimeout);
+      this.singleClickTimeout = null;
+    }
+  }
+});
+
+/**
+ * The base OpenTripPlanner map on which everything else is rendered.
+ */
+class BaseMap extends Component {
+  componentDidMount() {
+    const lmap = this.refs.map.leafletElement;
```

Comment:
See https://github.com/opentripplanner/otp-ui/pull/6/files#r359934885 about replacing `this.refs.map.leafletElement` with `this.leafletElement`.